test/Microsoft.Azure.SignalR.Tests.Common/TestRootHandler.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.SignalR.Tests.Common
+{
+    internal class TestRootHandler : HttpClientHandler
+    {
+        private readonly Action<HttpRequestMessage, CancellationToken> _callback;
+        private readonly HttpStatusCode _code;
+        private readonly string _content;
+
+        public TestRootHandler(HttpStatusCode code)
+        {
+            _code = code;
+        }
+
+        public TestRootHandler(HttpStatusCode code, string content) : this(code)
+        {
+            _content = content;
+        }
+
+        public TestRootHandler(Action<HttpRequestMessage, CancellationToken> callback)
```

Comment:
for this test handler, I think the basic need for its `.ctor` is to 
1. A way to assert if `HttpRequestMessage` is as expected
2. A way to construct the `HttpResponseMessage`

other `.ctor`s can be simplifications of the basic `.ctor`.